packages/wallet/src/KeyManagement/types.ts

```diff
@@ -13,32 +36,73 @@ export enum AddressType {
   External = 0
 }
 
-export enum KeyType {
-  External = 0,
-  Internal = 1,
-  Stake = 2
+export interface AccountAddressDerivationPath {
+  type: AddressType;
+  index: number;
 }
 
 export interface GroupedAddress {
-  networkId: Cardano.NetworkId;
   type: AddressType;
+  index: number;
+  networkId: Cardano.NetworkId;
   accountIndex: number;
-  addressIndex: number;
   address: Cardano.Address;
   rewardAccount: Cardano.RewardAccount;
 }
 
-export interface KeyManager {
-  // TODO: do not expose CSL objects publicly
-  // Reconsider which keys should be exposed, if any.
-  extendedAccountPublicKey: CSL.Bip32PublicKey;
-  derivePublicKey: (type: KeyType, index: number) => CSL.PublicKey;
-  // See https://github.com/cardano-foundation/CIPs/tree/master/CIP-1852#specification
-  deriveAddress: (type: AddressType, index: number) => GroupedAddress;
-  signMessage: (
-    addressType: AddressType,
-    signingIndex: number,
-    message: string
-  ) => Promise<{ publicKey: string; signature: string }>;
-  signTransaction: (tx: TxInternals) => Promise<Cardano.Witness['signatures']>;
+/**
+ * number[] is used by InMemoryKeyAgent
+ */
+export type AgentSpecificData = number[] | null;
+
+export interface SerializableKeyAgentDataBase {
+  networkId: Cardano.NetworkId;
+  accountIndex: number;
+}
+
+export interface SerializableInMemoryKeyAgentData extends SerializableKeyAgentDataBase {
+  __typename: KeyAgentType.InMemory;
+  encryptedRootPrivateKeyBytes: number[];
+}
+
+export interface SerializableLedgerKeyAgentData extends SerializableKeyAgentDataBase {
```

Comment:
Maybe drop this and all other Ledger types out until we have a Ledger implementation